fuse_ll/src/fuse/mount.rs

```diff
@@ -10,21 +11,158 @@ use std::path::Path;
 
 use param::*;
 
+pub struct MountOption<'a> {
+    pub name: &'a str,
+    // validator: fn(&str) -> bool,
+    pub parser: fn(&mut FuseMountArgs, &MountOption, &str),
+    pub regex: Regex,
+    #[cfg(target_os = "linux")]
+    pub flag: Option<u64>,
+    #[cfg(target_os = "macos")]
+    pub flag: Option<i32>,
+    #[cfg(target_os = "macos")]
+    pub fuse_flag: Option<u64>,
+}
+
+pub fn get_all_options() -> String {
+    let options: Vec<_> = get_mount_options()
+        .iter()
+        .map(|x| String::from(x.name))
+        .collect();
+    options.join(",")
+}
+
+fn add_option(options: &Option<String>, option: &str) -> Option<String> {
+    match options {
+        None => Some(String::from(option)),
+        Some(s) => Some(s.to_owned() + "," + option),
+    }
+}
+
+pub fn options_validator(option: String) -> Result<(), String> {
+    let ops: Vec<_> = option.split(',').collect();
+    let ret = ops
+        .iter()
+        .all(|&op| get_mount_options().iter().any(|x| x.regex.is_match(&op)));
+    if ret {
+        Ok(())
+    } else {
+        Err(format!(
+            "Invalid option \"{}\", valid options: {}",
+            option,
+            get_all_options()
+        ))
+    }
+}
+
 #[cfg(target_os = "linux")]
 mod param {
     // https://github.com/torvalds/linux/blob/master/include/uapi/linux/mount.h#L11
     // TODO: use mount flags from libc
-    // pub const MS_RDONLY: u64 = 1; // Mount read-only
+    pub const MS_RDONLY: u64 = 1; // Mount read-only
     pub const MS_NOSUID: u64 = 2; // Ignore suid and sgid bits
     pub const MS_NODEV: u64 = 4; // Disallow access to device special files
     pub const MNT_FORCE: i32 = 1; // Force un-mount
+
+    use super::MountOption;
+    use regex::Regex;
+    // Safe to use unwrap here becuase expression is always valid.
+    pub fn get_mount_options() -> Vec<MountOption<'static>> {
+        vec![
+            MountOption {
+                name: "ro",
+                parser: FuseMountArgs::parse_flag,
+                regex: Regex::new("^ro$").unwrap(),
```

Comment:
do not use unwrap() directly

Reply:
// Safe to use unwrap here becuase expression is always valid.
I have add comment at line 69. Because the regex is literal string and always valid at compile time. 
So I think it is OK to unwrap here. How do you think?